src/codeu/chat/server/Server.java

```diff
@@ -51,13 +55,22 @@
   private final Relay relay;
   private Uuid lastSeen = Uuids.NULL;
 
-  public Server(Uuid id, byte[] secret, Relay relay) {
+  private final Database database;
+  private final UserSchema userSchema;
+
+  public Server(Uuid id, byte[] secret, Relay relay, Database database) {
```

Comment:
Would be useful to add javadoc strings to these methods (I know these were given already, but still a good thing to add). :)